bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampMqttCommunication.java

```diff
@@ -0,0 +1,1028 @@
+/**
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ferroamp.internal;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ferroamp.dto.GetGeneralLx;
+import org.openhab.binding.ferroamp.dto.GetGeneralValues;
+import org.openhab.binding.ferroamp.dto.GetUdc;
+import org.openhab.core.io.transport.mqtt.MqttBrokerConnection;
+import org.openhab.core.io.transport.mqtt.MqttMessageSubscriber;
+import org.openhab.core.thing.Thing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+/**
+ * The {@link FerroampMqttCommunication} is responsible for communication with the Ferroamp-systems Mqtt-broker.
+ *
+ * @author Örjan Backsell - Initial contribution
+ *
+ */
+
+@NonNullByDefault
+public class FerroampMqttCommunication implements MqttMessageSubscriber {
+    static String[] ehubChannelsUpdateValues = new String[0];
+    static String[] ssoS1ChannelsUpdateValues = new String[0];
+    static String[] ssoS2ChannelsUpdateValues = new String[0];
+    static String[] ssoS3ChannelsUpdateValues = new String[0];
+    static String[] ssoS4ChannelsUpdateValues = new String[0];
+    static String[] esoChannelsUpdateValues = new String[0];
+    static String[] esmChannelsUpdateValues = new String[0];
+
+    static boolean isSsoChecked = false;
+    static String ssoS1IdCheck = "";
+    static String ssoS2IdCheck = "";
+    static String ssoS3IdCheck = "";
+    static String ssoS4IdCheck = "";
+
+    private final static Logger logger = LoggerFactory.getLogger(FerroampMqttCommunication.class);
+
+    public FerroampMqttCommunication(Thing thing) {
+        super();
+    }
+
+    // Handles request topic
+    static void sendPublishedTopic(String payload, FerroampConfiguration ferroampConfig) {
+        MqttBrokerConnection localConfigurationConnection = FerroampHandler.getFerroampConnection();
+        Objects.requireNonNull(localConfigurationConnection,
+                "MqttBrokerConnection localConfigurationConnection cannot be null");
+        localConfigurationConnection.start();
+        localConfigurationConnection.setCredentials(ferroampConfig.userName, ferroampConfig.password);
```

Comment:
Don't think you need to start the connection each time you publish something. It might be better to check if it is nonnected or not. If not => connect, otherwise just send the message.

Reply:
20250319
I will have a closer look, I will be back asap.